control-plane/agents/src/bin/core/volume/snapshot_operations.rs

```diff
@@ -553,4 +553,37 @@ impl OperationGuardArc<VolumeSnapshot> {
         }
         failed
     }
+
+    pub async fn pool_cluster_size_for_clone(&self, registry: &Registry) -> Result<u32, SvcError> {
```

Comment:
Also please add a comment explaining why is this needed now

```suggestion
    pub(crate) async fn pool_cluster_size_for_clone(&self, registry: &Registry) -> Result<u32, SvcError> {
```